### PR TITLE
Implemented IsHexDigit/FromHex in HttpEncoder

### PR DESCRIPTION
**Pull request by Yantrio**
Part of #2220, 
used an IfDef instead of replacement incase ms decides to update something.

### CODE REVIEW

**Comment by davidfowl** [comment]
Why ifdef at all? Just use the custom implementation.

**Comment by khellang** [comment]
Yeah, let's just keep the #ifdefs to an absolute minimum. If there's a bug in the implementation, at least it'll be consistent across platforms :wink:

**Comment by khellang** [comment]
:watch:

**Comment by Yantrio** [comment]
fixed.

File: src/Nancy/Helpers/HttpEncoder.cs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 //
 // Authors:
 //   Patrik Torstensson (Patrik.Torstensson@labs2.com)
-//   Wictor Wil�n (decode/encode functions) (wictor@ibizkit.se)
+//   Wictor Wil�n (decode/encode functions) (wictor@ibizkit.se)
```
Comment (khellang) [inline src/Nancy/Helpers/HttpEncoder.cs]: Pfff. Who needs tooling? :stuck_out_tongue_closed_eyes:

Comment (Yantrio) [inline src/Nancy/Helpers/HttpEncoder.cs]: i spent 1 morning with vs2015, and it did crap like this, im going back to atom.

**Pull request merged**